src/code/m_lights.c

```diff
@@ -0,0 +1,366 @@
+}
+
+void Global_light_list_ct(LightContext* lightCtx) {
+    lightCtx->list = NULL;
+}
+
+void func_8009BB40_jp(LightContext* lightCtx) {
+    while (lightCtx->list != NULL) {
+        Global_light_list_delete(lightCtx, lightCtx->list);
+        lightCtx->list = lightCtx->list->next;
+    }
+}
+
+LightNode* Global_light_list_new(UNUSED PlayState* play, LightContext* lightCtx, Lights* light) {
+    LightNode* newNode = Light_list_buf_new();
+
+    if (newNode != NULL) {
+        newNode->info = light;
+        newNode->prev = NULL;
+        newNode->next = lightCtx->list;
+        if (lightCtx->list != NULL) {
+            lightCtx->list->prev = newNode;
+        }
+        lightCtx->list = newNode;
+    }
+    return newNode;
+}
+
+void Global_light_list_delete(LightContext* lightCtx, LightNode* lightNode) {
+    if (lightNode != NULL) {
+        if (lightNode->prev != NULL) {
+            lightNode->prev->next = lightNode->next;
+        } else {
+            lightCtx->list = lightNode->next;
+        }
+        if (lightNode->next != NULL) {
+            lightNode->next->prev = lightNode->prev;
+        }
+        Light_list_buf_delete(lightNode);
+    }
+}
+
+LightsN* func_8009BC48_jp(GraphicsContext* gfxCtx, u8 ambient_r, u8 ambient_g, u8 ambient_b, u8 count, u8 light_r,
+                          u8 light_g, u8 light_b, s8 dir_x, s8 dir_y, s8 dir_z) {
+    LightsN* lights = GRAPH_ALLOC(gfxCtx, sizeof(LightsN) * 1);
+    s32 i;
+
+    lights->lights.a.l.col[0] = lights->lights.a.l.colc[0] = ambient_r;
+    lights->lights.a.l.col[1] = lights->lights.a.l.colc[1] = ambient_g;
+    lights->lights.a.l.col[2] = lights->lights.a.l.colc[2] = ambient_b;
+    lights->diffuse_count = count;
+
+    for (i = 0; i < count; i++) {
+        lights->lights.l[i].l.col[0] = lights->lights.l[i].l.colc[0] = light_r;
+        lights->lights.l[i].l.col[1] = lights->lights.l[i].l.colc[1] = light_g;
+        lights->lights.l[i].l.col[2] = lights->lights.l[i].l.colc[2] = light_b;
+
+        lights->lights.l[i].l.dir[0] = dir_x;
+        lights->lights.l[i].l.dir[1] = dir_y;
+        lights->lights.l[i].l.dir[2] = dir_z;
+    }
+
+    LightsN_disp(lights, gfxCtx);
+    return lights;
+}
+
+LightsN* new_LightsN(GraphicsContext* gfxCtx, u8 r, u8 g, u8 b) {
+    LightsN* lights = GRAPH_ALLOC(gfxCtx, sizeof(LightsN) * 1);
+
+    lights->lights.a.l.col[0] = r;
+    lights->lights.a.l.colc[0] = r;
+    lights->lights.a.l.col[1] = g;
+    lights->lights.a.l.colc[1] = g;
+    lights->lights.a.l.col[2] = b;
+    lights->lights.a.l.colc[2] = b;
+    lights->diffuse_count = 0;
+    return lights;
+}
+
+void Light_list_point_draw(PlayState* play) {
+    LightNode* lightNode = play->lightCtx.list;
+    Gfx* gfx;
+
+    OPEN_DISPS(play->state.gfxCtx);
+    gfx = func_800BD7C0_jp(POLY_XLU_DISP);
+
+    gDPSetAlphaDither(gfx++, G_AD_NOISE);
+    gDPSetColorDither(gfx++, G_CD_MAGICSQ);
+    gSPDisplayList(gfx++, D_400AA00);
```

Comment:
This dlist seems to be `point_light_init_model`, but we don't have a way to name objects' names, so leaving a comment with the name should be fine.